pcdsdevices/pseudopos.py

```diff
@@ -0,0 +1,83 @@
+from ophyd.device import Component as Cpt
+from ophyd.pseudopos import PseudoPositioner, PseudoSingle
+
+
+class SyncAxesBase(PseudoPositioner):
+    """
+    Synchronized Axes.
+
+    This will move all axes in a coordinated way, retaining offsets.
+
+    This can be configured to report its position as the min, max, mean, or any
+    custom function acting on a list of positions. Min is the default.
+
+    You should subclass this by adding real motors as components. The class
+    will pick them up and include them correctly into the coordinated move.
+
+    Parameters
+    ----------
+    position_mode: ``func``, optional
+        The function to apply to the list of current positions to determine
+        the combined position. By default, this is the minimum of all axis
+        positions.
+
+    Attributes
+    ----------
+    axes: ``list of str``
+        Names of the components that correspond to the real axes.
+    """
+    pseudo = Cpt(PseudoSingle)
+
+    _default_mode = min
```

Comment:
Any reason `min` would be the default? `mean` seems like the most intuitive to me but obviously an opinion.

Reply:
I originally had `mean` as the default but decided that it was conceptually easier to think of one motor being the "anchor" of the position and defining everything else relative to the motor, like the Beckhoff coupled axes with the master and slave axis. The `mean` is more nebulous as the position returned may not match any of the composite motors.

`min` vs `max` vs picking a random motor in the middle was pretty arbitrary.

Reply:
I see where you are coming from but I think there is a clear difference for this case. You are not explicitly labeling a leader and a follower. You'll read from one or the other based on position essentially swapping the "leader". I think unless you actually label one the follower `mean` is more intuitive.

Reply:
Strong disagree. I'd rather refactor so that motor #1 is the anchor value than swap from `min` to `mean`.

Reply:
e.g. I'd rather set it as

```
_default_mode = lambda pos: pos.__getitem__(0)
```

Reply:
That makes more sense to me than `min` ⬆️

Reply:
After thinking it over, I share you concerns with the arbitrary nature of `min`. I'll swap "first" to be the default behavior and adjust the tests accordingly.

I think `mean` makes sense for the detector tables fwiw, it's just the general case that has me worried.

Reply:
:+1: The important part is the user can pick with a line of Python code 😄

Reply:
Something "fun": built-ins attached to classes do not get passed "self", but all other functions do. So when I swapped `min` to `first` I got sad.

This will now be a subclass thing instead of a passable/class def override so that people can't make the mistake I just did.